features/workflow.feature

```diff
@@ -58,3 +64,55 @@ Feature: Workflow
         And the "BAZ" build succeeded
         Then the "XYZZY" job is started
         And that "XYZZY" build uses the same SHA as the "FOO" build
+
+    Scenario: Branch filtering (the pipeline's branch is committed)
+        Given an existing pipeline with the workflow:
+            | job   | requires        |
+            | FOO   | ~commit         |
+            | BAR   | ~commit:staging |
+            | BAZ   | ~commit:/^.*$/  |
+        When a new commit is pushed
+        And it is against the pipeline's branch
+        Then the "FOO" job is started
+        And the "BAR" job is not started
+        And the "BAZ" job is started
+
+    Scenario: Branch filtering (the staging branch is committed)
+        Given an existing pipeline with the workflow:
+            | job   | requires        |
+            | FOO   | ~commit         |
+            | BAR   | ~commit:staging |
+            | BAZ   | ~commit:/^.*$/  |
+        When a new commit is pushed
+        And it is against the staging branch
+        Then the "FOO" job is not started
+        And the "BAR" job is started
+        And the "BAZ" job is started
+
+    Scenario: Branch filtering (a pull request is opened to the pipeline's branch)
+        Given an existing pipeline with the workflow:
+            | job   | requires        |
+            | FOO   | ~pr             |
+            | BAR   | ~pr:staging     |
+            | BAZ   | ~pr:/^.*$/      |
+            | QUX   | ~pr:/master     |
+        When a pull request is opened
+        And it is targeting the pipeline's branch
+        Then the "FOO" job is started
+        And the "BAR" job is not started
+        And the "BAZ" job is started
+        And the "QUX" job is started
```

Comment:
"QUX" job is not started in current implementation.
ref: https://github.com/screwdriver-cd/screwdriver/blob/ba6478f5d43acae3fa1d5396cc2c417a591cee7a/plugins/webhooks/index.js#L176

Do we need to allow and work `~pr:<pipeline_branch>` ?